dialoptions.go

```diff
@@ -69,6 +74,7 @@ type dialOptions struct {
 	minConnectTimeout           func() time.Duration
 	defaultServiceConfig        *ServiceConfig // defaultServiceConfig is parsed from defaultServiceConfigRawJSON.
 	defaultServiceConfigRawJSON *string
+	mct                         time.Duration
```

Comment:
There is a `minConnectTimeout func() time.Duration` above - this would be duplicating that.

Also, in general: try not to use short names for things outside a small scope (i.e. a short function or a struct with 1-2 fields where it's entirely obvious what the short name means (e.g. timer.C)).